src/index.ts

```diff
@@ -146,40 +151,63 @@ function useDetectScroll(props: ScrollProps = {}): ScrollInfo {
 
   /** Function to update scroll direction */
   const updateScrollDir = useCallback(() => {
-    const scroll = axis === Axis.Y ? window.scrollY : window.scrollX;
+    const scroll =
+      target instanceof Window
+        ? axis === Axis.Y
+          ? window.scrollY
+          : window.scrollX
+        : axis === Axis.Y
+          ? target.scrollTop
+          : target.scrollLeft;
 
     if (Math.abs(scroll - lastScroll.current) >= threshold) {
       setScrollDir(scroll > lastScroll.current ? scrollDown : scrollUp);
       lastScroll.current = Math.max(0, scroll);
     }
     ticking.current = false;
-  }, [axis, threshold, scrollDown, scrollUp]);
+  }, [target, axis, threshold, scrollDown, scrollUp]);
 
   useEffect(() => {
     /** Function to update scroll position */
     const updateScrollPosition = () => {
-      const top = window.scrollY;
-      const left = window.scrollX;
+      const top = target instanceof Window ? target.scrollY : target.scrollTop;
```

Comment:
We may need to specify the `target` type here.

```suggestion
      const top = target instanceof Window ? (target as HTMLElement).scrollY : target.scrollTop;
```

Reply:
Good point! It was my bad. This would be the correct form:

```
const top = target instanceof Window ? target.scrollY : (target as HTMLElement).scrollTop;
```